### PR TITLE
Reorganization of the developers page index.md

### PR DESCRIPTION
**Pull request by CPSTL**
## Description
Reorganization of the Developers page on Ethereum.org

## How Has This Been Tested?
It should be further reviewed/discussed.

## Checklist:

- [x] I have read the contributing guidelines in the project README.
- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/developers/index.md [hunk 1]
```diff
@@ -16,336 +16,446 @@ sidebarDepth: 0
...
+**Looking for other options?**
 
+- [Ethereum Developers Tools List #SmartContractLanguages](https://github.com/ConsenSys/ethereum-developer-tools-list#smart-contract-languages)
 
 ## Language Specific Resources
 
 We're building a suite of language-specific landing pages for developer to learn about Ethereum in their preferred programming langauge.
 
-- [Ethereum for Java developers](/java)
+- [Ethereum for Java developers](https://ethereum.org/java/)
```
Comment (samajammin) [inline docs/developers/index.md]: Let's keep this link as just `/java` to avoid opening a new tab.

**Comment by jjmstark** [comment]
*Are* the items listed in the pre-requisites section really pre-requisites for all users of this page? For instance, does a developer building the simplest imaginable dapp need to understand merkle trees? 

Reason I ask is that I wouldn't want this list to scare developers away who don't feel they have enough experience to even try something very simple. 

If we do keep all these items, we should add appropriate links (e.g. to cryptography, hash functions, etc) so that we are also solving the problem for the user and helping them fill that knowledge gap.

**Comment by jjmstark** [comment]
Not sure if this is already captured by the fix you mention @samajammin (#203), but the weird Headers right now also mean that many of the category titles are not being grabbed for the nav bar as section titles.

**Comment by samajammin** [comment]
@jjmstark valid concern, we don't want to intimidate folks unnecessarily. I'm fine with removing the top 2 points (cryptography & merkle tree knowledge) or otherwise label them as a bonus/learning multiplier.

Regarding the headers not being grabbed, the [sidebar can be configured](https://v1.vuepress.vuejs.org/theme/default-theme-config.html#nested-header-links) to handle this in a nested format (if that's our ideal presentation, which I feel it should be). Here's what that could look like:

Image 2019-09-18 at 9 01 57 PM

Is that what you had in mind? If so I can push up that change.

**Comment by samajammin** [comment]
Spoke to @CPSTL & pushed those changes. @CPSTL @jjmstark take a look at the [deploy preview](https://deploy-preview-242--ethereumorg.netlify.com/developers/) & LMK if the sidebar menu, headers & prerequisites look good to ship or if you see any issues. Note that the artwork callout won't be an issue blocking the sidebar (addressed that in https://github.com/ethereum/ethereum-org-website/pull/243).

File: docs/developers/index.md [hunk 2]
```diff
@@ -9,334 +9,443 @@ meta:
...
 ---
 
 # Developer Resources
 
 <div class="featured">Guides, resources, and tools for developers building on Ethereum.</div>
 
+## Recommended Prerequisites
+
+- Familiarity with object-oriented programming languages.
+- [Understanding the principles of Blockchain](https://www.youtube.com/watch?v=SSo_EIwHSd4)
```
Comment (jjmstark) [inline docs/developers/index.md]:
```suggestion
- [Understanding the principles of Blockchain](https://www.youtube.com/watch?v=SSo_EIwHSd4) *Nov 13, 2017 - Simply Explained (Video)*
```

Comment (jjmstark) [reply docs/developers/index.md]: Keeping same format as other links below

File: docs/developers/index.md [hunk 3]
```diff
@@ -9,334 +9,443 @@ meta:
...
+- [Cryptoeconomics 101](https://www.youtube.com/watch?v=6RCtNrlzV28)
```
Comment (jjmstark) [inline docs/developers/index.md]:
```suggestion
- [Cryptoeconomics 101](https://www.youtube.com/watch?v=6RCtNrlzV28) *April 5, 2018 - Blockgeeks (Video)*
```

Comment (jjmstark) [reply docs/developers/index.md]: Keeping same format as other links below

**Comment by jjmstark** [comment]
Last question @CPSTL @samajammin - I don't think the two videos added to the prereq section were *previously* on the site, and want to make sure we've done a basic quality review. If you both think they're of sufficient quality / accuracy to serve the purpose of this page, then we're good to go imv.

**Pull request merged**